core/src/main/java/org/infinispan/metrics/impl/BaseOperatingSystemAdditionalMetrics.java

```diff
@@ -0,0 +1,105 @@
+package org.infinispan.metrics.impl;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+@NonNullApi
+@NonNullFields
+/**
+ * Provides process metrics.
+ * <p>
+ * Inspired by io.micrometer.core.instrument.binder.system.ProcessorMetrics
+ */
+public class BaseOperatingSystemAdditionalMetrics implements MeterBinder {
+
+   private static final Log log = LogFactory.getLog(BaseOperatingSystemAdditionalMetrics.class);
+
+   private final OperatingSystemMXBean operatingSystemBean;
+   private final Method processCpuLoadMethod;
+   private final Method processCpuTimeMethod;
+
+   public BaseOperatingSystemAdditionalMetrics() {
+      operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
+
+      Class<?> operatingSystemBeanClass = getOperatingSystemMXBeanImpl();
+      processCpuLoadMethod = detectMethod(operatingSystemBeanClass, "getProcessCpuLoad");
+      processCpuTimeMethod = detectMethod(operatingSystemBeanClass, "getProcessCpuTime");
+   }
+
+   @Override
+   public void bindTo(MeterRegistry registry) {
+      Gauge.builder("cpu.availableProcessors", operatingSystemBean, OperatingSystemMXBean::getAvailableProcessors)
+            .description("Displays the number of processors available to the Java virtual machine. This value may change during a particular invocation of the virtual machine.")
+            .register(registry);
+
+      Gauge.builder("cpu.systemLoadAverage", operatingSystemBean, OperatingSystemMXBean::getSystemLoadAverage)
+            .description("Displays the system load average for the last minute. The system load average is the sum of the number of runnable entities queued to the available processors and the number of runnable entities running on the available processors averaged over a period of time. The way in which the load average is calculated is operating system specific but is typically a damped time-dependent average. If the load average is not available, a negative value is displayed. This attribute is designed to provide a hint about the system load and may be queried frequently. The load average might be unavailable on some platforms where it is expensive to implement this method.")
+            .register(registry);
+
+      if (processCpuLoadMethod != null) {
+         Gauge.builder("cpu.processCpuLoad", operatingSystemBean, x -> invoke(processCpuLoadMethod))
```

Comment:
Looks like we aren't using the provided `operatingSystemBean` variable in the `ToDoubleFunction`, we can just use the `builder(String, Supplier)` method instead. Same with the next method.

Reply:
right!